src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -337,6 +337,13 @@ public static int CursorTop
         /// </summary>
         private static bool s_everReceivedCursorPositionResponse;
 
+        /// <summary>
+        /// Tracks if this is out first attempt to send a cursor posotion request. If it is, we start the
+        /// timer immediately (i.e. minChar = 0), but we use a slightly longer timeout to avoid the CPR response
+        /// being written to the console.
+        /// </summary>
+        private static bool s_firstCursorPositionRequest = true;
```

Comment:
You might consider an enum state machine instead of these bools.  Up to you (or whoever is primary owner of this codebase).

Here's a rough sketch.
```c#
private static CursorPositionResponseState s_cursorPositionResponseState = CursorPositionResponseState.NotSent;
private enum CursorPositionResponseState
{
    NotSent,
    NotReceieved,
    Recieved
};
...
if (s_cursorPositionResponseState == CursorPositionResponseState.NotSent)
{
    // allow for much longer timeout on first CPR
    Interop.Sys.InitializeConsoleBeforeRead(minChars: 0, decisecondsTimeout: 150);
    s_cursorPositionResponseState = CursorPositionResponseState.NotReceieved;
} 
else if (s_cursorPositionResponseState == CursorPositionResponseState.Recieved)
{
    // expect a response so don't start timer until 1 char is recieved
    Interop.Sys.InitializeConsoleBeforeRead(minChars: 1, decisecondsTimeout: 10);
}
else
{
    Interop.Sys.InitializeConsoleBeforeRead(minChars: 0, decisecondsTimeout: 10);
}
...
// we got a response, set state
s_everReceivedCursorPositionResponse = CursorPositionResponseState.Recieved;
```

Reply:
I'm OK sticking with the bools, the timing for setting `NotReceived` seems a bit unclear to me.